docs/advanced/performance.md

```diff
@@ -0,0 +1,67 @@
+# Performance Guide
+Roact operates on the principle that it's much easier to build UI declaratively. When something changes, the UI is rebuilt virtually and then the minimal necessary changes are made to the actual UI elements.
+
+For simple projects, performance is unlikely to be an issue. But Roact is built in Lua, and Lua can be slow. There are a number of techniques that you can do to avoid performance strain with your Roact apps.
+
+## Reduce Reconciliation
+
+### `shouldUpdate` Lifecycle Method
+When a Roact Component's state or props change, it will call the Component's `shouldUpdate` method to determine whether or not to re-render it. The default implementation will always returns true.
+```lua
+function Component:shouldUpdate(newProps, newState)
+	return true
+end
+```
+
+If you have a more complex component that only needs to re-render in certain situations, you can implement `shouldUpdate` and return `false` in any case where an update is not required.
```

Comment:
Might want to mention PureComponent before providing this alternative.

Reply:
We should also warn people that implementing `shouldUpdate` is _dangerous_!